src/core/server/http/lifecycle_handlers.ts

```diff
@@ -62,12 +62,12 @@ export const createVersionCheckPostAuthHandler = (kibanaVersion: string): OnPost
 };
 
 export const createCustomHeadersPreResponseHandler = (config: HttpConfig): OnPreResponseHandler => {
-  const serverName = config.name;
-  const customHeaders = config.customResponseHeaders;
+  const { name: serverName, securityResponseHeaders, customResponseHeaders } = config;
 
   return (request, response, toolkit) => {
     const additionalHeaders = {
-      ...customHeaders,
+      ...securityResponseHeaders,
+      ...customResponseHeaders,
       [KIBANA_NAME_HEADER]: serverName,
     };
```

Comment:
We explicitly allow `customResponseHeaders` to override any values in `securityResponseHeaders`. This is so the new config properties do not break existing configurations that already set the same headers using `customResponseHeaders`.

We could detect this situation when the config is set and throw an error, but we'd have to make that change in the 8..0 release, I think.